application.js

```diff
@@ -57,31 +54,59 @@ var sessionOptions = {
 
 // find out mongodb connection string from $fh.db
 function run(cb) {
-  sessionInit(sessionOptions, function(err) {
-    if (err) {
-      return cb(err);
-    }
-    // List the user fields which you don't want appearing in the authentication response.
-    // This is being consumed in the raincatcher-user mbaas router.
-    var authResponseExclusionList = ['password'];
-    raincatcherUser.init(mediator, app, authResponseExclusionList, sessionOptions, function(err) {
+  var port = process.env.FH_PORT || process.env.OPENSHIFT_NODEJS_PORT || 8001;
+  var host = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
+
+  // Check if direct mongo connection url exist (available after upgrading db) as mongoStore requires
+  // direct url and will throw an error when trying to connect.
+  if (! process.env.FH_MONGODB_CONN_URL) {
+    idleAndListen(port, host);
+  } else {
+    sessionInit(sessionOptions, function(err) {
       if (err) {
         return cb(err);
       }
-      require('./lib/user')(mediator);
+      // List the user fields which you don't want appearing in the authentication response.
+      // This is being consumed in the raincatcher-user mbaas router.
+      var authResponseExclusionList = ['password'];
+      raincatcherUser.init(mediator, app, authResponseExclusionList, sessionOptions, function(err) {
+        if (err) {
+          return cb(err);
+        }
+        require('./lib/user')(mediator);
 
-      // Important that this is last!
-      app.use(mbaasExpress.errorHandler());
-
-      var port = process.env.FH_PORT || process.env.OPENSHIFT_NODEJS_PORT || 8001;
-      var host = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
-      app.listen(port, host, function(err) {
-        cb(err, port);
+        // Important that this is last!
+        app.use(mbaasExpress.errorHandler());
+        app.listen(port, host, function(err) {
+          cb(err, port);
+        });
       });
     });
+  }
+}
+
+/**
+ * Idle and listen, leaves app running in order to allow database upgrade.
+ * @param port
+ * @param host
+ */
+function idleAndListen(port, host) {
+  console.log('*************************************************************************************');
+  console.log('Mongo Connection Error');
+  console.log('Please upgrade database in Data Browser part of the Studio and re-deploy your service');
+  console.log('MongoStore requires FH_MONGODB_CONN_URL system environment variable to be set,');
+  console.log('check Environment Variables part of the Studio');
+  console.log('*************************************************************************************');
+  app.use(mbaasExpress.errorHandler());
+  app.listen(port, host, function(err) {
+    if (err) {
+      console.error(err);
+      process.exit(1);
+    }
```

Comment:
If moving this to `lib/sessionInit`, to keep the node process running you could just use something like `function wait(){ setTimeout(wait, 9999999); }` recursion or just a blank express app or a plain [`http.Server`](https://nodejs.org/api/http.html#http_class_http_server) which could also return the error message via http.